zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -177,6 +183,78 @@ def _update_attribute(self, attrid, value):
             self.endpoint.electrical_measurement.power_factor_reported(value / 10)
 
 
+class ZemismartManufCluster(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Zemismart SPM01 Power Meter device."""
+
+    attributes = {
+        ZEMISMART_TOTAL_ENERGY_ATTR: ("energy", t.uint32_t, True),
+        ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR: ("reverse_energy", t.uint32_t, True),
+        ZEMISMART_VCP_ATTR: ("vcp_raw", t.data64, True),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == ZEMISMART_TOTAL_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_deliver_reported(value)
+        elif attrid == ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_receive_reported(value)
+        elif attrid == ZEMISMART_VCP_ATTR:
+            self.endpoint.electrical_measurement.voltage_reported(
+                (value[7] * 256) + value[6]
+            )
+            self.endpoint.electrical_measurement.current_reported(
+                (value[5] * 256 * 256) + (value[4] * 256) + value[3]
+            )
+            self.endpoint.electrical_measurement.power_reported(
+                (value[2] * 256 * 256) + (value[1] * 256) + value[0]
```

Comment:
Can you briefly explain what these values in the calculation are?

Reply:
The VCP data looks like this: [70, 1, 0, 249, 5, 0, 69, 9]
First three bytes are the power, next three are the current and the last two are the voltage
Voltage is multiplied by 10, current by 1000.
I don't know if there is a better way to do this.